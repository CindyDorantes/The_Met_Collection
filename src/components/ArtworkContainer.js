import React from 'react';

const ArtworkContainer = () => (
    <div>
        ArtworkContainer
    </div>
);

export default ArtworkContainer;