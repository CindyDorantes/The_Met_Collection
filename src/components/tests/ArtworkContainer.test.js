import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import store from '../../redux/configureStore';
import ArtworkContainer from '../ArtworkContainer';

describe('Artwork test', () => {
  test('snapshot for artwork', () => {
    const history = createBrowserHistory();
    const artwork = [{
      objectID: 300,
      primaryImage: 'https://images.metmuseum.org/CRDImages/ad/original/69178.jpg',
      primaryImageSmall: 'https://images.metmuseum.org/CRDImages/ad/web-large/69178.jpg',
      department: 'The American Wing',
      title: 'Balcony',
      culture: 'American',
      artistDisplayName: '',
      artistDisplayBio: '',
      objectDate: '1800 to 1830',
      medium: 'Wrought iron',
      dimensions: '44 1/2 x 242 in. (113 x 614.7 cm)',
      classification: '',
      objectURL: 'https://www.metmuseum.org/art/collection/search/300',
    }];

    history.push('/300', { artwork });
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router history={history}>
            <Routes>
              <Route path="/300" element={<ArtworkContainer props={artwork} />} />
            </Routes>
          </Router>
        </Provider>
      </React.StrictMode>,
    );
    expect(tree).toMatchSnapshot();
  });
});
