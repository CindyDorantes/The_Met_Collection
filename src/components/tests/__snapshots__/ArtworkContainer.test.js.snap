// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork test snapshot for artwork 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <p />
        <div
          class="mainContainer"
        >
          <div
            class="infoContainer"
          >
            <h1 />
            <h3 />
            <p />
          </div>
          <div
            class="imgContainer"
          >
            <img />
          </div>
        </div>
        <div
          class="mainContainer"
        >
          <h4>
            Artwork Details
          </h4>
          <div
            class="detailsContainer"
          >
            <div
              class="leftDetails"
            >
              <p>
                <span
                  class="bold"
                >
                  Title: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Artist: 
                </span>
                <span />
              </p>
              <p>
                <span
                  class="bold"
                >
                  Date: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Culture: 
                </span>
              </p>
            </div>
            <div
              class="rigthDetails"
            >
              <p>
                <span
                  class="bold"
                >
                  Medium: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Dimensions: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Classifications: 
                </span>
              </p>
              <p>
                <a>
                  Learn more...
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <p />
      <div
        class="mainContainer"
      >
        <div
          class="infoContainer"
        >
          <h1 />
          <h3 />
          <p />
        </div>
        <div
          class="imgContainer"
        >
          <img />
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <h4>
          Artwork Details
        </h4>
        <div
          class="detailsContainer"
        >
          <div
            class="leftDetails"
          >
            <p>
              <span
                class="bold"
              >
                Title: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Artist: 
              </span>
              <span />
            </p>
            <p>
              <span
                class="bold"
              >
                Date: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Culture: 
              </span>
            </p>
          </div>
          <div
            class="rigthDetails"
          >
            <p>
              <span
                class="bold"
              >
                Medium: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Dimensions: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Classifications: 
              </span>
            </p>
            <p>
              <a>
                Learn more...
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Coins test snapshot for coins 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <p />
        <div
          class="mainContainer"
        >
          <div
            class="infoContainer"
          >
            <h1 />
            <h3 />
            <p />
          </div>
          <div
            class="imgContainer"
          >
            <img />
          </div>
        </div>
        <div
          class="mainContainer"
        >
          <h4>
            Artwork Details
          </h4>
          <div
            class="detailsContainer"
          >
            <div
              class="leftDetails"
            >
              <p>
                <span
                  class="bold"
                >
                  Title: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Artist: 
                </span>
                <span />
              </p>
              <p>
                <span
                  class="bold"
                >
                  Date: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Culture: 
                </span>
              </p>
            </div>
            <div
              class="rigthDetails"
            >
              <p>
                <span
                  class="bold"
                >
                  Medium: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Dimensions: 
                </span>
              </p>
              <p>
                <span
                  class="bold"
                >
                  Classifications: 
                </span>
              </p>
              <p>
                <a>
                  Learn more...
                </a>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <p />
      <div
        class="mainContainer"
      >
        <div
          class="infoContainer"
        >
          <h1 />
          <h3 />
          <p />
        </div>
        <div
          class="imgContainer"
        >
          <img />
        </div>
      </div>
      <div
        class="mainContainer"
      >
        <h4>
          Artwork Details
        </h4>
        <div
          class="detailsContainer"
        >
          <div
            class="leftDetails"
          >
            <p>
              <span
                class="bold"
              >
                Title: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Artist: 
              </span>
              <span />
            </p>
            <p>
              <span
                class="bold"
              >
                Date: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Culture: 
              </span>
            </p>
          </div>
          <div
            class="rigthDetails"
          >
            <p>
              <span
                class="bold"
              >
                Medium: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Dimensions: 
              </span>
            </p>
            <p>
              <span
                class="bold"
              >
                Classifications: 
              </span>
            </p>
            <p>
              <a>
                Learn more...
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
